core/src/main/java/org/apache/iceberg/BaseRowDelta.java

```diff
@@ -104,6 +106,17 @@ public RowDelta validateNoConflictingDeleteFiles() {
 
   @Override
   protected void validate(TableMetadata base) {
+    // update startingSnapshotId.
```

Comment:
Changing the starting snapshot ID breaks the correctness of certain operations, so I don't think that's what you want to do. What are you trying to do here?